### ISSUE TITLE
First aggregate loses datetime precision

### ISSUE THREAD

**Issue by ahey** [ash-project/ash_postgres#176]
**Describe the bug**
When a first aggregate returns an attribute with type `:utc_datetime_usec`, precision is lost.

**To Reproduce**
[A failing test has been created to easily demonstrate the issue](https://github.com/ahey/ash_postgres/commits/first-aggregate-loss-of-precision).

```bash
1) test first aggregate maintains datetime precision (AshPostgres.AggregateTest)
     test/aggregate_test.exs:471
     Assertion with == failed
     code:  assert post.created_at == fetched_post.latest_comment_created_at
     left:  ~U[2023-11-13 03:46:05.332378Z]
     right: ~U[2023-11-13 03:46:05Z]
     stacktrace:
       test/aggregate_test.exs:494: (test)
```

The bug seems to have been introduced in `ash` [by these changes](https://github.com/ash-project/ash/compare/d9bacc15548ce8f91086fccff03048812e1469ab...7849541b2995be6a91b4f62355b9c6eb7e54e32a).

Here is [the most recent `ash` commit where the issue was not present](https://github.com/ash-project/ash/commit/f89219296946258127e7491f6829e21d2e12fe74)

**Comment by zachdaniel** [comment]
Hmm...yeah this looks like an issue that needs resolving. We need to get aggregate constraints as well as types here.

**Comment by zachdaniel** [comment]
This is fixed in 2.17

**Comment by ahey** [comment]
Thanks Zach!
